### PR TITLE
Add codelab kubernetes operator codelab

### PR DESCRIPTION
**Pull request by shwetamazumder**

### CODE REVIEW

**Comment by shwetamazumder** [comment]
This is ready to go live. Can we have someone to review and approve this?

**Comment by shwetamazumder** [comment]
@paul-kondrat - Can you add your review comments here?

**Review by paul-kondrat — APPROVED** [review]
Looks good to me.

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
+author: shwetamazumder
+summary:
+id: kubernetes_operator_codelab
+tags:
+categories:
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:5]: Can you please add `Kubernetes` in the categories section

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+# How to deploy Solace Pubsub+ Event Broker with Pubsub+ Event Broker Operator in Kubernetes
+
+## What you'll learn: Overview
+
+Duration: 0:01:00
+
+Hello and Welcome! If you are wondering how to use our Operator , then you are in the right place! 
+
+This codelab will show how to:
+* Setup a kubernetes environment with Kind
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:19]: I would suggest adding a hyperlink to Kind here as well

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+* Install the Solace Pubsub+ Event Broker Operator in kubernetes
+* Deploy a dev instance of Solace Pubsub+ Event Broker using the Operator
+* Test the Pubsub+ Event Broker deployement and login to Pubsub+ Manager
+
+
+<!--### Info Boxes
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:25]: No need for this comment section

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+We need to make sure the below installations are complete and successful before we create a cluster using Kind:
+* Install [Docker Desktop](https://docs.docker.com/desktop/install/mac-install/) on macOS. For other OS options you can visit [Docker](https://docs.docker.com/get-docker/).
+* Install Kubernetes command-line tool [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/) on macOS. For other OS options you can visit: \
+            [kubectl for Linux](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/)\
+            [kubectl for Windows](https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/) 
+
+
+
+
+<!--### Add a Link
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:68]: Same note on comment section

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+## Prerequisites
+
+Duration: 0:05:00
+
+You will need a Kubernetes Cluster to deploy the Operator. There are several options available on Windows, Mac and Linux to setup a Kubernetes Cluster such as, [Docker Desktop](https://docs.docker.com/desktop/kubernetes/) , [Rancher Desktop](https://rancherdesktop.io/), [kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/), [minikube](https://minikube.sigs.k8s.io/docs/start/) etc.
+
+I will be using [kind](https://kind.sigs.k8s.io/) to create my Kubernetes Cluster for today's tutorial. It creates the cluster which spins up Docker Containers acting as nodes. 
+
+Note: Kind installation is covered in the next step. If you already have a Kubernetes Cluster on your machine then you can directly skip to Step 4: Install Solace Pubsub+ Event Broker Operator.
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:57]: Can we add this in a "Positive" styled format as follows
`<aside class="positive"> Kind installation is covered.... </aside>`

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+kind create cluster
+```
+### 3. Verify that the Kubernetes Environment is ready
+```bash
+# This command returns the list of worker nodes and their status
+kubectl get nodes
+```
+![Environment Setup Verification](img/Operator_Img2.png)
+
+### Next Step is to install the Operator
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:113]: No need for this H3

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+
+# Wait for deployment to complete
+kubectl get pods -n pubsubplus-operator-system --watch
+```
+Note: If you do not have SSH setup on your machine and got an error cloning the repo, you can clone the https link instead `https://github.com/SolaceProducts/pubsubplus-kubernetes-operator.git`
+
+
+![Operator Setup Verification](img/Operator_Img3.png)
+
+### Next Step is to install the Event Broker
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:145]: No need for this H3

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+git clone git@github.com:SolaceProducts/pubsubplus-kubernetes-operator.git
+
+# cd to the pubsubplus-kubernetes-operator/deploy folder and apply deploy.yaml
+# Manifest creates a namespace and all K8s resources for the Operator deployment
+kubectl apply -f deploy.yaml
+
+# Wait for deployment to complete
+kubectl get pods -n pubsubplus-operator-system --watch
+```
+Note: If you do not have SSH setup on your machine and got an error cloning the repo, you can clone the https link instead `https://github.com/SolaceProducts/pubsubplus-kubernetes-operator.git`
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:140]: I would suggest adding the same aside format here as before
`<aside class="positive"> If you do not have SSH .... </aside>`

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+# Wait for broker deployment pods to be ready
+kubectl get pods --show-labels --watch
+
+# Check service-ready
+kubectl wait --for=condition=ServiceReady eventbroker dev-example
+```
+
+![Broker Setup Verification](img/Operator_Img4.png)
+
+### Next Step is to test the deployment
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:176]: No need for this H3

Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:176]: No need for this H3

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+```
+Note: If you do not have SSH setup on your machine and got an error cloning the repo, you can clone the https link instead `https://github.com/SolaceProducts/pubsubplus-kubernetes-operator.git`
+
+
+![Operator Setup Verification](img/Operator_Img3.png)
+
+### Next Step is to install the Event Broker
+
+## Install Solace Pubsub+ Event Broker 
+
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:148]: Missing `Duration` block under this title. 

Can you please add the following under this header
`Duration: 0:03:00`

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+kubectl wait --for=condition=ServiceReady eventbroker dev-example
+```
+
+![Broker Setup Verification](img/Operator_Img4.png)
+
+### Next Step is to test the deployment
+
+## Test Solace Pusbub+ Event Broker Deployment
+
+Duration: 0:05:00
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:180]: I believe this step is more than 5 minutes... can we increase this to 15 minutes

File: markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+Duration: 0:01:00
+
+
+This codelab shows how the operator makes an event broker deployment easy by simply applying the yaml file. We have deployed a dev instance of the Pubsub+ Event Broker using the Operator in a Kubernetes cluster, where, Kind is used to run Kubernetes locally on a mac machine. \
+Checkout [Kubernetes Operator - Quickstart](https://github.com/SolaceProducts/pubsubplus-kubernetes-quickstart) for more deployment options.
+<!--✅ < Fill IN TAKEAWAY 1>   
+✅ < Fill IN TAKEAWAY 2>   
+✅ < Fill IN TAKEAWAY 3>   -->
+
+![Soly Image Caption](img/soly.gif)
```
Comment (TamimiGitHub) [inline markdown/kubernetes_operator_codelab/kubernetes_operator_codelab.md:241]: In the preview link, looks like this image is broken hmm. Did you run the `export.sh` script and made sure all the files are added and committed? 
https://644fe87763f0753cfd819359--codelabs-solace.netlify.app/codelabs/kubernetes_operator_codelab/index.html?index=..%2F..index#6

Screen Shot 2023-05-05 at 6 46 10 PM

**Comment by TamimiGitHub** [comment]
Thanks for your contribution! I added a couple of comments

**Comment by shwetamazumder** [comment]
I have made the changes . @TamimiGitHub

**Review by TamimiGitHub — APPROVED** [review]
Looks good to me! can be merged. Thanks again for your contribution :)

**Pull request merged**